broker/src/main/java/io/zeebe/broker/system/partitions/impl/steps/AtomixLogStoragePartitionStep.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.system.partitions.impl.steps;
+
+import io.zeebe.broker.system.partitions.PartitionContext;
+import io.zeebe.broker.system.partitions.PartitionStep;
+import io.zeebe.logstreams.storage.atomix.AtomixLogStorage;
+import io.zeebe.util.sched.future.ActorFuture;
+import io.zeebe.util.sched.future.CompletableActorFuture;
+
+public class AtomixLogStoragePartitionStep implements PartitionStep {
+
+  @Override
+  public ActorFuture<Void> open(final PartitionContext context) {
+    final var openFuture = new CompletableActorFuture<Void>();
+    final var server = context.getRaftPartition().getServer();
+
+    final var appenderOptional = server.getAppender();
+    appenderOptional.ifPresentOrElse(
+        logAppender -> {
```

Comment:
What if this `logAppender` is already from a newer term? This is not checked else where, right? So, in theory, it is still possible for a LogstorageAppender from an older term to write to a LeaderRole at newer term.

Reply:
That check doesn't prevent this, right? It is only checking if there was an error on notifying listeners and there is already a new role, then ignore the error. Because the new transition might succeed.

Do you think it is going to be a big change to make it a followup issue? IMO without checking the term, the bug is not really fixed.

Reply:
Lets talk about via zoom